Add a cleanup timeout to the run as me Process implementation

tobwiens opened the pull request:
Add a cleanup timeout to the run as me Process implementation

Problem:
- Run as me Process implementation does SIGKILL whole process tree

Solution:
- Retrieve the cleanup timeout and pass it to the kill_process_tree.sh script
- kill_process_tree.sh SIGTERMS the process tree, then waits for the graceful task termination timeout. After that it sigterms the process tree.
- The kill_process_tree.sh waits half of the graceful task termination termination timeout
- command_step.sh and kill_process_tree.sh have SIGTERM traps, to not be killed when SIGTERM is received.

fviale commented on programming-extensions/programming-extension-processbuilder/src/test/java/org/objectweb/proactive/extensions/processbuilder/LinuxProcessBuilderTest.java:
```diff
@@ -110,6 +110,32 @@ public void su_destroy_process() throws Exception {
+    @Test(timeout = 15000)
+    public void su_destroy_process_sigkill_after_graceful_timeout() throws Exception {
+        LinuxProcessBuilder processBuilder = new LinuxProcessBuilder(new OSUser(username, password), null,
+                proactiveHome);
```
shouldn't you set proactive.process.builder.cleanup.time.seconds=2 to avoid waiting for a long period in the test ?

fviale commented on programming-extensions/programming-extension-processbuilder/src/main/resources/processbuilder/linux/kill_process_tree.sh:
```diff
@@ -4,12 +4,73 @@
 # pattern. Any process which does not belong the current user will be ingored.
 
 LOGIN=$(whoami)
+
```
clean up this file

fviale reviewed (changes requested):
a minor request but overall it is fine

fviale commented on programming-extensions/programming-extension-processbuilder/src/main/resources/processbuilder/linux/kill_process_tree.sh:
```diff
@@ -4,12 +4,73 @@
 for ppid in `pgrep -u $LOGIN -f $1`;
 do
-  if [ $ppid -ne $$ ]
+  if [ $ppid -ne $$ -a $ppid -ne $parentPid ]
```
this succession of for, if, for is very difficult to understand. Could you create a function to make things a bit clearer

tobwiens committed:
Add a cleanup timeout to the run as me Process implementation

Problem:
- Run as me Process implementation does SIGKILL whole process tree

Solution:
- Retrieve the cleanup timeout and pass it to the kill_process_tree.sh script
- kill_process_tree.sh SIGTERMS the process tree, then waits for the graceful task termination timeout. After that it sigterms the process tree.
- The kill_process_tree.sh waits half of the graceful task termination termination timeout
command_step.sh and kill_process_tree.sh have SIGTERM traps, to not be killed when SIGTERM is received.
